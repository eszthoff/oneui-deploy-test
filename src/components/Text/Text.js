import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './Text.scss';

const { block } = bem({
    name: 'Text',
    classnames: styles,
    propsToMods: ['muted']
});

const Text = props => {
    const { children, inline, ...rest } = props;
    const HtmlNodeType = inline ? 'span' : 'p';

    return (
        <HtmlNodeType {...rest} {...block(props)}>
            {children}
        </HtmlNodeType>
    );
};

Text.displayName = 'Text';

Text.propTypes = {
    /** Text should be rendered inline */
    inline: PropTypes.bool,
    /** Render muted (auxiliary) text */
    muted: PropTypes.bool,
    /** Text content */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired
};

Text.defaultProps = {
    inline: false,
    muted: false
};

export default Text;
