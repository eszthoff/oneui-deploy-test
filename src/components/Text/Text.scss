@import '../../themes/nice/nice.scss';

.Text {
    color: var(--color-foreground);
    font: {
        family: var(--font-family-primary);
        size: var(--font-size-normal);
        weight: var(--font-weight-normal);
    }

    &--muted {
        color: var(--color-muted);
    }
}
