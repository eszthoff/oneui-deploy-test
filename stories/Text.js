import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { boolean, text, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
import Text from '../src/components/Text';

storiesOf('Text', module)
    .addDecorator(withKnobs)
    .add('Text', () => (
        <Text inline={boolean('Inline text', false)} muted={boolean('Muted text', false)}>
            {text('Content', 'This is some text content')}
        </Text>
    ));
