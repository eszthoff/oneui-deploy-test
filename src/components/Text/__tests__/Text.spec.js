import React from 'react';
import toJson from 'enzyme-to-json';
import Text from '../Text';

describe('<Text> that renders a text block', () => {
    it('should render default text block correctly', () => {
        const wrapper = shallow(<Text>Some text content</Text>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should add classes when props are changed', () => {
        const wrapper = shallow(
            <Text inline muted>
                Some inline, muted text content
            </Text>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
