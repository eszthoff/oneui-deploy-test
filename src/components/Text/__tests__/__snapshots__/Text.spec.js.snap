// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> that renders a text block should add classes when props are changed 1`] = `
<span
  className="Text Text--muted"
  muted={true}
>
  Some inline, muted text content
</span>
`;

exports[`<Text> that renders a text block should render default text block correctly 1`] = `
<p
  className="Text"
  muted={false}
>
  Some text content
</p>
`;
